File: data/compilers_comparison/src/examples/second_example/src/func54.c
#include "second_example.h" 
array_t* func54(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array33 = func55(&params0, loopsFactor);
      DEBUG_RETURN(array33->id);
      free(params0.data);
      array33->refC--;
      if(array33->refC == 0) {
         free(array33->data);
         free(array33);
         DEBUG_FREE(array33->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array35 = func56(&params0, loopsFactor);
      DEBUG_RETURN(array35->id);
      free(params0.data);
      array35->refC--;
      if(array35->refC == 0) {
         free(array35->data);
         free(array35);
         DEBUG_FREE(array35->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array37 = func57(&params0, loopsFactor);
   DEBUG_RETURN(array37->id);
   free(params0.data);
   return array37;
}

File: data/multilang_comparison/program_nim/src/func54.nim
proc func54*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array52: ptr ArrayT = func55(addr params0, loopsFactor)
      debugReturn(array52.id)
      dealloc(params0.data)
      dec array52.refC
      if array52.refC == 0:
         dealloc(array52.data)
         dealloc(array52)
         debugFree(array52.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array54: ptr ArrayT = func56(addr params0, getPath(), loopsFactor)
      debugReturn(array54.id)
      dealloc(params0.data)
      dec array54.refC
      if array54.refC == 0:
         dealloc(array54.data)
         dealloc(array54)
         debugFree(array54.id)
   var params0: ArrayTParam
   params0.size = 0
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   var array58: ptr ArrayT = func57(addr params0, loopsFactor)
   debugReturn(array58.id)
   dealloc(params0.data)
   return array58

